packages/service-library/src/servicelib/volumes_utils.py

```diff
@@ -0,0 +1,54 @@
+from pathlib import Path
+from typing import Final, Optional
+
+import aiofiles
+from pydantic import BaseModel, Field, root_validator
+
+HIDDEN_FILE_NAME: Final[str] = ".hidden_do_not_remove"
+AGENT_FILE_NAME: Final[str] = ".agent"
+
+
+class VolumeState(BaseModel):
+    requires_saving: bool = Field(
+        ...,
+        description=(
+            "when True, the contents of the volume must be saved whe "
+            "closing the service"
+        ),
+    )
+    was_saved: Optional[bool] = Field(
+        None,
+        description=(
+            "When `True`: the sidecar managed to save the contents of the volume. "
+            "When `False`: there was an issue and the contes was not saved, it is "
+            "up to the agent to save the content of the volume. "
+            "When `None`: if requires_saving is False, it makes more sense "
```

Comment:
now reading that description I start understand it.
so requires_saving is the flag that defines whether saving will happen or not. Whatever the value of was_saved is. just wondering would it not make sense to have ```last_saved: Optional[datetime]``` ? it is always only once. then maybe not.